ra/ra.go

```diff
@@ -864,7 +865,8 @@ func (ra *RegistrationAuthorityImpl) recheckCAA(ctx context.Context, authzs []*c
 		// If the result had a CAA boulder error, construct a suberror with the
 		// identifier from the authorization that was checked.
 		if err := recheckResult.err; err != nil {
-			if bErr, _ := err.(*berrors.BoulderError); berrors.Is(err, berrors.CAA) {
+			var bErr *berrors.BoulderError
+			if errors.As(err, &bErr); berrors.Is(err, berrors.CAA) {
```

Comment:
I might rewrite this slightly as:

```suggestion
			if errors.As(err, &bErr) && bErr.Type == berrors.CAA {
```

The role of `berrors.Is` is to (a) unwrap and (b) check the Type field. Since we already need to do the unwrap in order to append a SubBoulderError below, we may as well rely on the unwrapped error and check the Type field ourselves. This winds up with something that looks like a more typical `errors.As` pattern and saves the reader some effort.